document/json.go

```diff
@@ -117,12 +121,34 @@ func (vb *ValueBuffer) UnmarshalJSON(data []byte) error {
 
 // ToJSON encodes d to w in JSON.
 func ToJSON(w io.Writer, d Document) error {
-	return json.NewEncoder(w).Encode(jsonDocument{d})
+	buf, err := jsonDocument{d}.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	_, err = w.Write(buf)
+	if err != nil {
+		return err
+	}
+
+	_, err = w.Write([]byte{'\n'})
```

Comment:
Why do we need to add a `\n` at the end of the buffer? Same for `ArrayToJSON` function.

Reply:
Great question, I'll add a comment

Reply:
I have removed it because it didn't really make sense anymore. We'd better have this function return a json object without newline, and rely on `json.Encoder` if we want to control the behaviour.